### DISCUSSION TITLE
Supported OpenIDC grant/flow?

### DISCUSSION THREAD

**Discussion by holtgrewe** [fastapi-users/fastapi-users#1304] [Q&A] [answered]
Hello.

I'm trying to integrate with LifeScience-RI which allows EU academics to authenticate with their home organisations credentials. When trying to setup an account, I was asked which flow/grant to use and I specified "implicit".

Here is a list of OpenIDC flows

https://darutk.medium.com/diagrams-of-all-the-openid-connect-flows-6968e3990660

The service desk wrote

> while processing the request we noticed that the user requested the Implicit grant. We are planning to drop the option and support for it. The implicit flow is known to be insecure, it had been deprecated and it is now removed from latest OAuth2 specs.
>
> Does the service really need that grant? If not we would prefer to remove it and proceed with the other options, such as the recommended authorization_code.

So now my question is which flows/grants are supported by fastapi-users?

Thanks

**Comment by frankie567** [comment, marked as answer]
FastAPI Users doesn't directly handle OAuth2 request, [`httpx-oauth`](https://github.com/frankie567/httpx-oauth) does. If you use the base `OAuth2` client, it'll use authorization code grant: https://github.com/frankie567/httpx-oauth/blob/2e82654559b1687a6b25c86e31dc9290ae06cdba/httpx_oauth/oauth2.py#L105-L109

**Reply by holtgrewe** [reply]
thanks
